lib/vector-core/src/event/metric/tags.rs

```diff
@@ -216,32 +256,19 @@ impl Iterator for TagValueIter {
     }
 }
 
-impl<'a> IntoIterator for &'a TagValueSet {
-    type IntoIter = TagValueRefIter<'a>;
-    type Item = &'a str;
-
-    fn into_iter(self) -> Self::IntoIter {
-        match self {
-            TagValueSet::Single(tag) => TagValueRefIter::Single(tag.as_ref()),
-            TagValueSet::Set(set) => TagValueRefIter::Set(set.into_iter()),
-        }
-    }
-}
-
 pub enum TagValueRefIter<'a> {
-    Single(Option<&'a String>),
+    Single(Option<&'a TagValue>),
     Set(<&'a IndexSet<TagValue> as IntoIterator>::IntoIter),
 }
 
 impl<'a> Iterator for TagValueRefIter<'a> {
-    type Item = &'a str;
+    type Item = Option<&'a str>;
```

Comment:
For the sake clarity: would it/could it make sense to also have `type TagValueRef<'a> = Option<&'a str>`? As the companion of `TagValue`.

Reply:
I thought about that, but wasn't sure how useful it would be. Will add.